### PR TITLE
add prosemirror-history redo/undo plugin

### PR DESCRIPTION
**Pull request by xiadd**
By the way, Why still keep .yarn in the repo? it's too large

### CODE REVIEW

File: packages/core/src/editor.ts [hunk 1]
```diff
@@ -151,8 +152,12 @@ export class Editor {
                 if (!mark) throw new Error();
                 return (cur as Required<Mark>).keymap(mark);
             });
+        const historyKeymap = {
+            'Mod-z': undo,
+            'Shift-Mod-z': redo,
+        };
 
-        return [...nodesKeymap, ...marksKeymap].map((keys) => keymap(keys));
+        return [...nodesKeymap, ...marksKeymap, historyKeymap].map((keys) => keymap(keys));
```
Comment (Saul-Mirone) [inline packages/core/src/editor.ts]: Move this line to 184 would be better

**Comment by Saul-Mirone** [comment]
> By the way, Why still keep .yarn in the repo? it's too large

Because at first I want to try the yarn zero install mode...

File: packages/core/src/editor.ts (hunk 1 above)
Comment (xiadd) [reply packages/core/src/editor.ts]: image

You mean here?

Comment (Saul-Mirone) [reply packages/core/src/editor.ts]: Yes

**Pull request merged**